cypress/integration/tests/reports/pagination.test.ts

```diff
@@ -0,0 +1,273 @@
+/// <reference types="cypress" />
+
+import { login, clickByText, selectItemsPerPage } from "../../../utils/utils";
+import { navMenu } from "../../views/menu.view";
+import { reports, applicationinventory } from "../../types/constants";
+import { ApplicationInventory } from "../../models/applicationinventory/applicationinventory";
+import * as data from "../../../utils/data_utils";
+import {
+    selectItemsPerPageAdoptionCandidate,
+    selectItemsPerPageIdentifiedRisks,
+    expandArticle,
+} from "../../models/reports/reports";
+import { Stakeholders } from "../../models/stakeholders";
+import * as commonView from "../../views/common.view";
+
+var applicationsList: Array<ApplicationInventory> = [];
+var stakeholdersList: Array<Stakeholders> = [];
+
+describe("Reports pagination validations", () => {
+    before("Login and create test data", function () {
+        // Perform login
+        login();
+
+        const stakeholder = new Stakeholders(data.getEmail(), data.getFullName());
+        stakeholder.create();
+        stakeholdersList.push(stakeholder);
+
+        // Navigate to Application inventory tab
+        clickByText(navMenu, applicationinventory);
+        var rowsToCreate = 0;
+
+        // Get the current table row count and create appropriate test data rows
+        selectItemsPerPage(100);
+        cy.get(commonView.appTable)
+            .next()
+            .then(($div) => {
+                if (!$div.hasClass("pf-c-empty-state")) {
+                    cy.get(".pf-c-table > tbody > tr")
+                        .not(".pf-c-table__expandable-row")
+                        .find("td[data-label=Name]")
+                        .then(($rows) => {
+                            var rowCount = $rows.length;
+                            if (rowCount <= 10) {
+                                rowsToCreate = 10 - rowCount;
+                            }
+                            if (rowsToCreate > 0) {
+                                // Create multiple Applications
+                                for (let i = 0; i < rowsToCreate; i++) {
+                                    const application = new ApplicationInventory(
+                                        data.getFullName()
+                                    );
+                                    application.create();
+                                    applicationsList.push(application);
+                                }
```

Comment:
Can we have this for block as a common function, just to avoid same code replication from line 60